htsinfer/extract_random_seq_subset.py

```diff
@@ -0,0 +1,324 @@
+"""Extract a random subset of the sample data."""
+
+import shutil
+import random
+import gzip
+import logging
+from typing import List, Tuple, IO
+from enum import Enum
+from Bio.SeqIO.QualityIO import FastqGeneralIterator  # type: ignore
+
+
+class Outcomes(Enum):
+    """Enumerator for processing outcomes."""
+    files_verified = "files_verified"
+    no_input_provided = "no_input_provided"
+    no_output_provided = "no_output_provided"
+    no_input_output_provided = "no_input_output_provided"
+    mismatched_in_out_files = "mismatched_in_out_files"
+    same_in_out_files = "same_in_out_files"
+    extraction_done = "extraction_done"
+    file_error = "file_error"
+    no_input_sequences = "no_input_sequences"
+    invalid_number_requested = "invalid_number_requested"
+    unexpected_return_value = "unexpected_return_value"
+
+
+LOGGER = logging.getLogger(__name__)
+
+
+def extract(
+    infile: List[str],
+    outfile: List[str],
+    proportion: float = 1.0,
+        max_records: int = 10000) -> str:
+
+    """Extract a subset of reads from one or two FASTQ files.
+
+    Args:
+        infile (List[str]) : List of file paths to read libraries
```

Comment:
When using type hints, it's better not to add the type info to the docstrings, as autodoc tools parse the type info from the type hints and stating it in two places is a source of errors/confusion